### PR TITLE
Removed use of deprecated SYCL_DEVICE_FILTER

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Replaced it with `ONEAPI_DEVICE_SELECTOR` environment variable.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

File: examples/python/device_selection.py [hunk 1]
```diff
@@ -25,8 +25,8 @@ def create_default_device():
     Create default SyclDevice using `sycl::default_selector`.
 
     Device created can be influenced by environment variable
-    SYCL_DEVICE_FILTER, which determines SYCL devices seen by the
-    SYCL runtime.
+    ONEAPI_DEVICE_SELECTOR, which determines SYCL devices seen by the
```
Comment (antonwolfy) [inline examples/python/device_selection.py:28]: `SYCL_DEVICE_FILTER` still mentioned in `docs/_legacy/docfiles/user_guides/manual/dpctl/device_selection.rst` and `docs/_legacy/docfiles/user_guides/manual/dpctl/devices.rst` files.
Is it expected?

File: examples/python/device_selection.py [hunk 2]
```diff
@@ -25,8 +25,8 @@ def create_default_device():
...
+    oneAPI DPC++ runtime.
```
Comment (antonwolfy) [inline examples/python/device_selection.py:29]: Should be `SYCL runtime` replaced with `oneAPI DPC++ runtime` everywhere? Still present in many files.

File: examples/python/device_selection.py (hunk 1 above)
Comment (oleksandr-pavlyk) [reply examples/python/device_selection.py:28]: `docs/_legacy` contains sources of `dpctl` from before gh-1619. This folder will be removed in near future.

File: examples/python/device_selection.py (hunk 2 above)
Comment (oleksandr-pavlyk) [reply examples/python/device_selection.py:29]: Probably yes, it is more precise, as we do not presently support other SYCL implementations, but I would defer that change to a different PR.

**Review by antonwolfy — APPROVED** [review]
No more comments from me. Thank you @oleksandr-pavlyk

**Pull request merged**